specs/agents/otel-distribution.md

```diff
@@ -0,0 +1,81 @@
+
+## Terminology
+
+**Vanilla OpenTelemetry Distribution**: this is the "upstream" OpenTelemetry distribution that is maintained by the OpenTelemetry community.
+Implementation differs per platform, but it usually consists of an API/SDK and can also provide automatic instrumentation.
+
+**Elastic OpenTelemetry Distribution**: this is an OpenTelemetry distribution provided by Elastic that is derived from the 
+_Vanilla OpenTelemetry Distribution_.
+
+## General guidelines
+
+These statements are guiding principles of the Elastic OpenTelemetry distributions, they should be considered more as advice than strict rules.
+
+Elastic OpenTelemetry distribution SHOULD ideally:
+- behave as drop-in replacements of their upstream counterparts
+- provide a simple setup and favor onboarding experience (aka "things should work by default").
+- avoid capturing potentially confusing data (see [system metrics](#system-metrics) example below).
+
+## Configuration
+
+Elastic OpenTelemetry distributions MAY override the default configuration.
+When doing so, user-configuration should remain consistent with vanilla distribution:
+- explicit user configuration SHOULD remain effective
+- overriden default configuration MUST have the ability to be restored to upstream default
+
+Elastic specific configuration items MUST be prefixed with `ELASTIC_OTEL_`.
+For example, the [universal profiling integration](#universal-profiling-integration) can be enabled with `ELASTIC_OTEL_UNIVERSAL_PROFILING_INTEGRATION_ENABLED`.
+
+Elastic and platform specific configuration items must be prefixed with `ELASTIC_OTEL_${platform}_` to be consistent with
+the upstream `OTEL_${platform}_` prefix.
+
+When introducing new features, the decision between starting with platform-specific or general namespace is made on a feature by feature case:
+- feature can be aligned cross-platform even if implemented only in only one: use `ELASTIC_OTEL_` prefix, for example [System metrics](#system-metrics).
+- feature that we know will be platform-specific: use `ELASTIC_OTEL_${platform}_` prefix.
```

Comment:
I have clarified here the strategy to "favor general over platform" when we can, let me know if that's not clear enough.

The pros of this strategy:
- good for cross-platform configuration, at least to enable common features
- forces us to think about cross-platform implementation
- migrating from one platform implementation to more does not forces us to deal with supporting both general and platform config level for compatibility

downsides:
- we might have configuration at two levels for a given feature as the platform-level details will be in a different level as the general ones.

Reply:
LGTM